src/backend/joanie/core/flows/order.py

```diff
@@ -25,92 +24,166 @@ def _set_order_state(self, value):
     def _get_order_state(self):
         return self.instance.state
 
-    def _can_be_state_submitted(self):
+    def _can_be_assigned(self):
         """
-        An order can be submitted if the order has a course, an organization,
-        an owner, and a product
+        An order can be assigned if it has an organization.
+        """
+        return self.instance.organization is not None
+
+    @state.transition(
+        source=enums.ORDER_STATE_DRAFT,
+        target=enums.ORDER_STATE_ASSIGNED,
+        conditions=[_can_be_assigned],
+    )
+    def assign(self, billing_address=None):
+        """
+        Transition order to assigned state.
+        """
+        if not self.instance.is_free and billing_address:
+            Address = apps.get_model("core", "Address")  # pylint: disable=invalid-name
+            address, _ = Address.objects.get_or_create(
+                **billing_address,
+                owner=self.instance.owner,
+                defaults={
+                    "is_reusable": False,
+                    "title": f"Billing address of order {self.instance.id}",
+                },
+            )
```

Comment:
I wonder if we should check that `billing_address` is set when order is not free.

Reply:
Yes, but maybe in OrderViewSet.create ?

Reply:
We are already checking that on the `OrderViewSet` but I'm comfortable with that as we could trigger this transition out of the scope of this viewset then we will not check if billing address is defined and transition will succeed with incomplete.

Reply:
TODO added